### PR TITLE
Add globalization

### PR DESCRIPTION
**Pull request by superkhau**
Connect to strongloop/loopback/issues/2422

### CODE REVIEW

**Comment by superkhau** [comment]
@slnode test please

**Comment by 0candy** [comment]
LGTM

**Comment by superkhau** [comment]
@slnode test please

**Comment by superkhau** [comment]
@0candy @loay PTAL

**Comment by 0candy** [comment]
LGTM :+1:

File: lib/discovery.js [hunk 1]
```diff
@@ -175,15 +177,15 @@ function mixinDiscovery(Oracle) {
    */
   function getArgs(table, options, cb) {
     if ('string' !== typeof table || !table) {
-      throw new Error('table is a required string argument: ' + table);
+      throw new Error(g.f('table is a required string argument: %s', table));
```
Comment (loay) [inline lib/discovery.js]: {{table}}

File: lib/discovery.js [hunk 2]
```diff
@@ -175,15 +177,15 @@ function mixinDiscovery(Oracle) {
...
     }
     options = options || {};
     if (!cb && 'function' === typeof options) {
       cb = options;
       options = {};
     }
     if (typeof options !== 'object') {
-      throw new Error('options must be an object: ' + options);
+      throw new Error(g.f('options must be an object: %s', options));
```
Comment (loay) [inline lib/discovery.js]: {{options}}
not sure about {{object}} though

File: lib/transaction.js [hunk 3]
```diff
@@ -22,7 +24,7 @@ function mixinTransaction(Oracle, oracle) {
     debug('Begin a transaction with isolation level: %s', isolationLevel);
     if (isolationLevel !== Transaction.READ_COMMITTED &&
       isolationLevel !== Transaction.SERIALIZABLE) {
-      var err = new Error('Invalid isolationLevel: ' + isolationLevel);
+      var err = new Error(g.f('Invalid isolationLevel: %s', isolationLevel));
```
Comment (loay) [inline lib/transaction.js]: {{isolationLevel}}

**Comment by loay** [comment]
@superkhau reviewed. 
LGTM for the single use convention.
I found couple of other words to be exempted from translation, so up to you and @0candy  here in case you have double checked with @thejsoto

File: lib/discovery.js (hunk 2 above)
Comment (superkhau) [inline lib/discovery.js]: both are valid TY

**Comment by superkhau** [comment]
@0candy @loay Thanks for the feedback, I've applied them all. Gonna merge.

**Comment by superkhau** [comment]
Unrelated CI failures. Merging.

**Pull request merged**